lib/datadog/di/component.rb

```diff
@@ -53,12 +57,12 @@ def environment_supported?(settings)
           # TODO add tests?
           unless settings.dynamic_instrumentation.internal.development
             if Datadog::Core::Environment::Execution.development?
-              Datadog.logger.debug("Not enabling dynamic instrumentation because we are in development environment")
+              Datadog.logger.warn("Not enabling dynamic instrumentation because we are in development environment")
               return false
             end
           end
           if RUBY_ENGINE != 'ruby' || RUBY_VERSION < '2.6'
-            Datadog.logger.debug("Not enabling dynamic instrumentation because of unsupported Ruby version")
+            Datadog.logger.warn("Not enabling dynamic instrumentation because of unsupported Ruby version")
```

Comment:
For this one, I suggest being a bit more actionable, e.g. 

```suggestion
            Datadog.logger.warn("Cannot enable dynamic instrumentation: Ruby 2.6+ is required")
```